### ISSUE TITLE
Staged Builders should work with ImplementationVisibility.PRIVATE (to hide Immutable* and use *Builder)

### ISSUE THREAD

**Issue by vorburger** [immutables/immutables#470]
Upon using the new staged builders introduced by @elucash in https://github.com/immutables/immutables/issues/438 (FYI @androidfred @agebhar1), which are very cool, I've noticed a combination of stagedBuilder together with ImplementationVisibility.PRIVATE which could use an improvement / is broken, try this:

```
@Immutable
@Value.Style(stagedBuilder = true, visibility = ImplementationVisibility.PRIVATE)
abstract class Example {

    abstract int number();

}
```

This generates:

`public final class ExampleBuilder implements NumberBuildStage, BuildFinal {`

but.. there are no `NumberBuildStage` & `BuildFinal` generated!

**Comment by vorburger** [comment]
I've found this work around (?) for now:

@Immutable
@Value.Style(stagedBuilder = true)
abstract class Example {

    abstract int number();

    public static class Builder extends Example.Builder { }
}

**Comment by elucash** [comment]
As this functionality was created extremely quickly, no wonder that not all are cases working ) I'll take a look

**Comment by elucash** [comment]
addressed (in duplicate #761) by disabling this combination, basically, I think this combination doesn't provide much value as outer builder exposes full builder API, unrestricted by build interfaces. (closing this as duplicate, but it's actually the other way around ))
